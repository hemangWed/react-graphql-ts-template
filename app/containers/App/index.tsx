/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */
import React from 'react';

import { routeConfig } from '@app/routeConfig';
import { Layout } from 'antd';
import map from 'lodash-es/map';
import { withRouter } from 'react-router';
import { Route, Switch } from 'react-router-dom';
import { compose } from 'redux';
import styled, { ThemeProvider } from 'styled-components';
import For from '@components/For';
import Header from '@components/Header';
import { colors } from '@themes/index';
import Sidebar from '@app/components/Siderbar';
import '@app/styles/index.css';
const theme = {
  fg: colors.primary,
  bg: colors.secondaryText
};

const CustomLayout = styled(Layout)`
  && {
    flex-direction: row;
  }
`;

export function App() {
  return (
    <ThemeProvider theme={theme}>
      <Header />
      <CustomLayout>
        <Sidebar />
        <span className="text-3xl font-bold underline">Hello world!</span>
        <Layout.Content>
          <For
            ParentComponent={(props) => <Switch {...props} />}
            of={map(Object.keys(routeConfig))}
            renderItem={(routeKey, index) => {
              const Component = routeConfig[routeKey].component;
              return (
                <Route
                  exact={routeConfig[routeKey].exact}
                  key={index}
                  path={routeConfig[routeKey].route}
                  render={(props) => {
                    const updatedProps = {
                      ...props,
                      ...routeConfig[routeKey].props
                    };
                    return <Component {...updatedProps} />;
                  }}
                />
              );
            }}
          />
        </Layout.Content>
      </CustomLayout>
    </ThemeProvider>
  );
}

export default compose(withRouter)(App);
