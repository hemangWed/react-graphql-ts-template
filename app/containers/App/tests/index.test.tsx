import React from 'react';
import { renderProvider } from '@utils/testUtils';
import App from '@containers/App';
import { BrowserRouter } from 'react-router-dom';

describe('<App /> container tests', () => {
  it('should render and match the snapshot', () => {
    const { container } = renderProvider(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
