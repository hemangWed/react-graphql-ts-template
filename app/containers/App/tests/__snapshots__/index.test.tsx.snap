// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> container tests should render and match the snapshot 1`] = `
<div>
  <header
    class="ant-layout-header Header__StyledHeader-a7tfu0-0 eDpQQE"
    data-testid="header"
  >
    <a
      href="/"
    >
      <img
        alt="logo"
        class="Header__Logo-a7tfu0-1 hbWjcY"
        src="IMAGE_MOCK"
      />
    </a>
    <p
      class="T__StyledText-gcrjip-0 emthK Header__Title-a7tfu0-2 elUMIm"
      data-testid="t"
    >
      Wednesday Solutions
    </p>
  </header>
  <section
    class="ant-layout App__CustomLayout-di9tar-0 gvzqHR"
  >
    <div
      class="Siderbar__SidebarWrapper-qw2kux-0 gaTYUT"
    >
      <div
        class="Siderbar__SideBarStatic-qw2kux-2 lhyiGU"
        data-testid="sidebar"
      >
        <a
          aria-label="home link"
          data-testid="rocket-home-link"
          href="/"
        >
          <img
            alt="rocket-icon"
            class="Siderbar__RocketLogo-qw2kux-3 iVxHgo"
            src="IMAGE_MOCK"
          />
        </a>
      </div>
    </div>
    <main
      class="ant-layout-content"
    />
  </section>
</div>
`;
